test/Haechi/13.sol

```diff
@@ -0,0 +1,193 @@
+
+        // deploy mock oracle factory for deployed token (named gold)
+        oracleFactory = new MockOracleFactory();
+        oracles.push(
+            oracleFactory.newOracle(
+                address(tokens[0]),
+                "goldOracle",
+                9,
+                20000000000
+            )
+        );
+        oracles.push(IOracle(Mainnet.CHAINLINK_EUR_USD));
+        oracles.push(IOracle(Mainnet.CHAINLINK_CAD_USD));
+        oracles.push(IOracle(Mainnet.CHAINLINK_USDC_USD));
+
+        // deploy new assimilator factory & curveFactory v2
+        assimFactory = new AssimilatorFactory();
+        curveFactory = new CurveFactoryV2(
+            50000,
+            address(accounts[2]),
+            address(assimFactory)
+        );
+        assimFactory.setCurveFactory(address(curveFactory));
+        // now deploy curves
+        cheats.startPrank(address(accounts[2]));
+        for (uint256 i = 0; i < 3; ++i) {
+            CurveInfo memory curveInfo = CurveInfo(
+                string(abi.encode("dfx-curve-", i)),
+                string(abi.encode("lp-", i)),
+                address(tokens[i]),
+                address(tokens[3]),
+                DefaultCurve.BASE_WEIGHT,
+                DefaultCurve.QUOTE_WEIGHT,
+                oracles[i],
+                tokens[i].decimals(),
+                oracles[3],
+                tokens[3].decimals(),
+                DefaultCurve.ALPHA,
+                DefaultCurve.BETA,
+                DefaultCurve.MAX,
+                DefaultCurve.EPSILON,
+                DefaultCurve.LAMBDA
+            );
+            Curve _curve = curveFactory.newCurve(curveInfo);
+            curves.push(_curve);
+        }
+        cheats.stopPrank();
+
+        // now mint gold & silver tokens
+        uint256 mintAmt = 300_000_000_000;
+        for (uint256 i = 0; i < 4; ++i) {
+            decimals.push(utils.tenToPowerOf(tokens[i].decimals()));
+            if (i == 0) {
+                tokens[0].mint(address(accounts[0]), mintAmt.mul(decimals[i]));
+            } else {
+                deal(
+                    address(tokens[i]),
+                    address(accounts[0]),
+                    mintAmt.mul(decimals[i])
+                );
+            }
+        }
+        // now approve
+        cheats.startPrank(address(accounts[0]));
+        for (uint256 i = 0; i < 3; ++i) {
+            tokens[i].approve(address(curves[i]), type(uint256).max);
+            tokens[3].approve(address(curves[i]), type(uint256).max);
+        }
+        cheats.stopPrank();
+    }
+
+    function testMinDeposit() public {
+        uint256 amt = 1000000;
+        // mint token to trader
+        deal(address(tokens[1]), address(accounts[1]), amt * decimals[1]);
+
+        uint256 eurocToSend = 1000000;
+        // mint tokens to attacker
+        deal(address(tokens[1]), address(curves[1]), eurocToSend);
+        deal(address(tokens[3]), address(curves[1]), eurocToSend * 100);
```

Comment:
Here we're actually seeding the LP, right?

Reply:
it's a bit different from seeding what we used to do after new pool deployment. This time, we are directly sending tokens to the pool with 100:1 ratio and this doesn't mint any LPTs.

Reply:
Ah ok I see, the comment threw me off. Bit of a nitpick, but we should can change or delete it if it's not accurate. Just a matter of safety for naive folks like me looking at the code and perhaps assuming the wrong thing.